### PR TITLE
Introduce support for new Query DSL #263

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: src/main/resources/lib/guillotine/generic/enum-types.js [hunk 1]
```diff
@@ -24,6 +24,59 @@ function createEnumTypes(context) {
             'inline': 'inline'
         }
     });
+
+    context.types.dslQueryType = context.schemaGenerator.createEnumType({
+        name: context.uniqueName('DslQueryType'),
+        description: 'Query Dsl type.',
```
Comment (vbradnitski) [inline src/main/resources/lib/guillotine/generic/enum-types.js:30]: `name - DslQueryType`
`description - Query Dsl type`

but 

`name - DslOperatorType`
`description - Dsl Operator Type`

It's better to have the same order everywhere

**Review by vbradnitski — CHANGES_REQUESTED** [review]

File: src/main/resources/lib/guillotine/util/factory.js [hunk 2]
```diff
@@ -76,5 +76,273 @@ function processBooleanFilter(inputFilter) {
...
+        holder.like = dslExpression;
+    }
+}
+
+function createDslRangeExpression(holder, expression) {
+    if (expression) {
+        let rangeExpression = {
+            field: expression.field,
+        };
+        if (expression.lt) {
```
[the remark this answers is not in the document]
Comment (anatol-sialitski) [reply src/main/resources/lib/guillotine/util/factory.js:160]: @vbradnitski no issue here. The validation done [here](https://github.com/enonic/lib-guillotine/blob/9fcf2d057a684b53d4274a6b6e954dc34aefe47b/src/main/resources/lib/guillotine/util/validation.js#L92)

File: src/main/resources/lib/guillotine/generic/enum-types.js (hunk 1 above)
Comment (anatol-sialitski) [reply src/main/resources/lib/guillotine/generic/enum-types.js]: Done

File: src/main/resources/lib/guillotine/util/factory.js [hunk 3]
```diff
@@ -76,5 +76,273 @@ function processBooleanFilter(inputFilter) {
...
+        return graphQLValue.localDateTimeValue;
+    }
+    if (graphQLValue.localTimeValue) {
+        return graphQLValue.localTimeValue;
+    }
+    if (graphQLValue.instantValue) {
+        return graphQLValue.instantValue;
+    }
+    throw 'Value must be not null';
+}
```
Comment (ComLock) [inline src/main/resources/lib/guillotine/util/factory.js:318]: So we're happy with this? Lets say someone creates a range with lessthan string, greaterthan double? I'm assuming the validator will make a nice error message?

**Comment by ComLock** [comment]
Looks like the future is bright https://stepzen.com/blog/coming-soon-to-graphql-the-oneof-input-object

**Comment by ComLock** [comment]
https://github.com/graphql/graphql-spec/pull/825

**Comment by ComLock** [comment]
https://github.com/graphql/graphql-wg/blob/main/rfcs/Struct.md#composite-type-capable-of-input-polymorphism

**Review by ComLock — CHANGES_REQUESTED** [review]

**Comment by ComLock** [comment]
Although I personally like going from "doubleValue" to just "double", I'm going to point out something, just so it's thought about. The words "double" and "boolean" (haven't checked the rest) are reserved words in JavaScript.
https://www.w3schools.com/js/js_reserved.asp

This means that they cannot be used as variables, labels, or function names.

So if one wants to deconstruct such a property from an object, one would have to rename the variable in the decontruct:

```
const { double: myDouble } = object;
```

**Review by rymsha — APPROVED** [review]

File: src/main/resources/lib/guillotine/util/factory.js (hunk 3 above)
Comment (anatol-sialitski) [reply src/main/resources/lib/guillotine/util/factory.js]: I added validation for that

**Review by rymsha — APPROVED** [review]

**Review by ComLock — APPROVED** [review]

**Pull request merged**